src/domain/usecases/ImportTemplateUseCase.ts

```diff
@@ -109,7 +123,27 @@ export class ImportTemplateUseCase implements UseCase {
 
         const importResult = await this.instanceRepository.importDataPackage(dataValues);
 
-        return Either.success(_.compact([deleteResult, ...importResult]));
+        const importResultHasErrors = importResult.flatMap(result => result.errors);
+        if (template.type === "custom" && (importResultHasErrors.length > 0 || deleteResult)) {
+            const orgUnitName = removeCharacters(
+                await this.excelRepository.readCell(templateId, template.fixedOrgUnit, {
+                    formula: false,
+                })
+            );
+
```

Comment:
You are getting the org unit from the template, as we discussed, that's a good option to avoid a request. The thing is, by doing it this way, we are limiting to custom templates, right? (`if (template.type === "custom"`) I wonder if it would be better to have these org units and make it work for all types. 

Also, I am thinking about the general feature. In the error, we have in the ID of the entity that is causing the problem: `importResult[0].errors[0].id` -> `hmZE3mVAZFf`.  

So my proposal (check with the project manager if unsure), would be, instead of parsing specific errors (there may be more to what you matched), just get the referenced ID in the error. If it's found in the list of org units, we know we want to set the details collumn. "Message" column is quite explanatory, the only thing missing is the name. So:

Message = Organisation unit: `hmZE3mVAZFf` not in hierarchy of current user: `OqlIhNLAJXo`
Details = `Organisation unit: BurkinaFaso`

What do you think?

To get the organisation units, this is working for me:

```
if (!dataForm) {
    return Either.error({ type: "DATA_FORM_NOT_FOUND" });
}

const orgUnits = await this.instanceRepository.getDataFormOrgUnits(dataForm.type, dataFormId);
```

Reply:
Thank you @tokland , and thank you @eperedo for the code. Arnau, I'm sorry but the problem is that Bulk Load users are typically users without any skill in DHIS2 and that not necessarily understand its jargon and logic. For them, "OU xxxx not in hierarchy of current user yyyy" does not have anything to do with "You are trying to submit data for a OU you are not assigned to". I understand your point that translating DHIS2 messages is a titanic task, and I think the intention is not to translate any possible message, but just there were we can, do it, and there were we cannot, just show DHIS2 message. 

My proposal was actually something like. When there is an alternative explanation that we know, present that alternative message, showing in details the DHIS2 message. And when it is not present, just show the DHIS2 response as message. 

Does it make sense?

Reply:
@ifoche Yes, we can keep the matching for known messages. My doubt was what should we do when there is no match, but we know that `error.id` is a org unit ID. Do we show some "details"? (if yes, what literal exactly?)

Reply:
The literal from @eperedo is ok. I like it more than mine as he's showing which OU you cannot write in. 

As for the situation of receiving a OU ID without any matching. I think that "Details" is perfectly fine. That's included right? I'm merging assuming that's the case. We can open a new task if it is not and you think it is important